Base the defalt window screen and height off the desktop size.

mplooster opened the pull request:
Seems useful if you want to have the window scale to the size of the desktop (which I don't know if you do or not).

mplooster committed:
Base the defalt window screen and height off the desktop size.

RasmusD reviewed (changes requested):
That is indeed not a bad idea. However on linux at least, due to the way windows snap they don't actually use the full screen real estate and this puts some of the sides beyond the window borders.
I suggest setting it to:

desktop.width * 0.95f and desktop.height * 0.95f

This will resolve the issue for now. There's a TODO to make screen resolution configurable and enable full screen etc. for the future.

But this was a good test to see if things are scaling right - which they seem t so far!

mplooster committed:
Set the default screen size to 0.95 of the available size.

This is to account for a weird sfml behavior where the returned
desktop size does not account for task bars (particularly on
ubuntu linux).

mplooster commented:
Sorry for the delay in updating this. I was trying my best to see if there was some way to account for the window snapping behavior, but I couldn't find anything quick and easy (seems like this is something SFML should handle for us, but it is what it is).

RasmusD commented:
No worries. No rush.
